src/Roots.jl

```diff
@@ -285,14 +271,48 @@ function make_rhs(model::RootsModel)
         dY.vegetation.rwc[1] = flow_out_roots - flow_out_stem
         dY.vegetation.rwc[2] =
             flow_out_stem -
-            compute_transpiration(model.configuration,t)
+            compute_transpiration(model.configuration.transpiration, t)
     end
     return rhs!
 end
 
+
+
+
+
+abstract type AbstractRootExtraction{FT <: AbstractFloat} end
+abstract type AbstractTranspiration{FT <: AbstractFloat} end
+
+"""
+    RootsConfiguration{FT} <: AbstractConfiguration{FT}
+
+The Configuration for running the Roots model. 
+
+Both transpiration and root extraction model types must be provided.
+These types are using for dispatching when computing the RHS.
+"""
+struct RootsConfiguration{
+    FT,
+    RET <: AbstractRootExtraction{FT},
+    TT <: AbstractTranspiration{FT},
+} <: AbstractConfiguration{FT}
+    root_extraction::RET
+    transpiration::TT
+end
+
+struct PrognosticSoilPressure{FT} <: AbstractRootExtraction{FT} end
```

Comment:
More nitpicking - is calling this `Prognostic` misleading here? It is a prognostic variable for the soil model, but not for the root model.

Reply:
I appreciate nitpicking!

It's a good point. Perhaps, since we discussed moving these types and methods to the LSM repo eventually, since it is only in the LSM case that they are needed (rather than having them be in the component modules), I will leave it with this name. Then, when we move it, it is consistent that SoilPressure is a prognostic variable, since it is for the LSM....

Reply:
That's a good point!